Panic during AWS scanning of API Gateway

## Description

I'm using trivy of main branch. When I scan AWS, I got the following error.


```bash
> git log -n 1
commit d343d13ac6005a48f637cb9adb4390402695feac (HEAD -> main, origin/main, origin/HEAD)
Author: Liam Galvin <liam.galvin@aquasec.com>
Date:   Tue Aug 16 13:00:18 2022 +0100

    fix: upgrade defsec to v0.71.7 for elb scan panic (#2720)

> make build
go build -ldflags "-s -w -X=main.version=0.30.0-73-gd343d13a " ./cmd/trivy

> ./trivy aws --region ap-northeast-1
[1/31] Scanning api-gateway...
[2/31] Scanning athena...
[3/31] Scanning cloudfront...
[4/31] Scanning cloudtrail...
[5/31] Scanning cloudwatch...
[6/31] Scanning codebuild...
[7/31] Scanning documentdb...
[8/31] Scanning dynamodb...
[9/31] Scanning ec2...
[10/31] Scanning ecr...
[11/31] Scanning ecs...
[12/31] Scanning efs...
[13/31] Scanning eks...
[14/31] Scanning elasticache...
[15/31] Scanning elasticsearch...
[16/31] Scanning elb...
[17/31] Scanning emr...
[18/31] Scanning iam...
[19/31] Scanning kinesis...
[20/31] Scanning kms...
[21/31] Scanning lambda...
[22/31] Scanning mq...
[23/31] Scanning msk...
[24/31] Scanning neptune...
[25/31] Scanning rds...
[26/31] Scanning redshift...
[27/31] Scanning s3...
[28/31] Scanning sns...
[29/31] Scanning sqs...
[30/31] Scanning ssm...
[31/31] Scanning workspaces...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x79e7bba]

goroutine 1 [running]:
github.com/aquasecurity/defsec/internal/rules/aws/apigateway.glob..func1(0x0)
	/Users/saso/go/pkg/mod/github.com/aquasecurity/defsec@v0.71.7/internal/rules/aws/apigateway/enable_access_logging.go:39 +0x5a
github.com/aquasecurity/defsec/internal/rules.RegisteredRule.Evaluate({0x0, {{0xaf49e1f, 0xc}, {0x0, 0x0, 0x0}, {0xafb010b, 0x15}, {0xb0e64d3, 0x43}, ...}, ...}, ...)
	/Users/saso/go/pkg/mod/github.com/aquasecurity/defsec@v0.71.7/internal/rules/register.go:25 +0x3b
github.com/aquasecurity/defsec/pkg/scanners/cloud/aws.(*Scanner).Scan(0xc0022961e0, {0xc018160, 0xc002194840})
	/Users/saso/go/pkg/mod/github.com/aquasecurity/defsec@v0.71.7/pkg/scanners/cloud/aws/scanner.go:101 +0x42a
github.com/aquasecurity/trivy/pkg/cloud/aws/scanner.(*AWSScanner).Scan(_, {_, _}, {{{0xaf38694, 0xa}, 0x0, 0x0, 0x0, 0x0, 0x34630b8a000, ...}, ...})
	/Users/saso/.ghq/github.com/aquasecurity/trivy/pkg/cloud/aws/scanner/scanner.go:60 +0x59a
github.com/aquasecurity/trivy/pkg/cloud/aws/commands.Run({_, _}, {{{0xaf38694, 0xa}, 0x0, 0x0, 0x0, 0x0, 0x34630b8a000, {0xc003bca480, ...}, ...}, ...})
	/Users/saso/.ghq/github.com/aquasecurity/trivy/pkg/cloud/aws/commands/run.go:146 +0x3ce
github.com/aquasecurity/trivy/pkg/commands.NewAWSCommand.func2(0xc00215bb80, {0xc004617d80?, 0x2?, 0x2?})
	/Users/saso/.ghq/github.com/aquasecurity/trivy/pkg/commands/app.go:844 +0x1bb
github.com/spf13/cobra.(*Command).execute(0xc00215bb80, {0xc004617d60, 0x2, 0x2})
	/Users/saso/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc001fd6500)
	/Users/saso/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/saso/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.run()
	/Users/saso/.ghq/github.com/aquasecurity/trivy/cmd/trivy/main.go:37 +0x17c
main.main()
	/Users/saso/.ghq/github.com/aquasecurity/trivy/cmd/trivy/main.go:19 +0x19
```

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

## What did you expect to happen?


## What happened instead?


## Output of run with `-debug`:

```
2022-08-16T22:12:01.717+0900	DEBUG	[defsec] 12:01.717013000 aws-api.scanner                  There were 1 errors during adaption process: failed to run adapter for api-gateway: operation error ApiGatewayV2: GetDomainNames, https response error StatusCode: 200, RequestID: 8d77a773-1d30-46e9-8bcd-811d4db6efa0, deserialization failed, failed to decode response body with invalid JSON, expected __timestampIso8601 to be of type string, got json.Number instead
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x79e7bba]

goroutine 1 [running]:
github.com/aquasecurity/defsec/internal/rules/aws/apigateway.glob..func1(0x0)
	/Users/saso/go/pkg/mod/github.com/aquasecurity/defsec@v0.71.7/internal/rules/aws/apigateway/enable_access_logging.go:39 +0x5a
github.com/aquasecurity/defsec/internal/rules.RegisteredRule.Evaluate({0x0, {{0xaf49e1f, 0xc}, {0x0, 0x0, 0x0}, {0xafb010b, 0x15}, {0xb0e64d3, 0x43}, ...}, ...}, ...)
	/Users/saso/go/pkg/mod/github.com/aquasecurity/defsec@v0.71.7/internal/rules/register.go:25 +0x3b
github.com/aquasecurity/defsec/pkg/scanners/cloud/aws.(*Scanner).Scan(0xc0021f6000, {0xc018160, 0xc00305c780})
	/Users/saso/go/pkg/mod/github.com/aquasecurity/defsec@v0.71.7/pkg/scanners/cloud/aws/scanner.go:101 +0x42a
github.com/aquasecurity/trivy/pkg/cloud/aws/scanner.(*AWSScanner).Scan(_, {_, _}, {{{0xaf38694, 0xa}, 0x0, 0x0, 0x1, 0x0, 0x34630b8a000, ...}, ...})
	/Users/saso/.ghq/github.com/aquasecurity/trivy/pkg/cloud/aws/scanner/scanner.go:60 +0x59a
github.com/aquasecurity/trivy/pkg/cloud/aws/commands.Run({_, _}, {{{0xaf38694, 0xa}, 0x0, 0x0, 0x1, 0x0, 0x34630b8a000, {0xc003602480, ...}, ...}, ...})
	/Users/saso/.ghq/github.com/aquasecurity/trivy/pkg/cloud/aws/commands/run.go:146 +0x3ce
github.com/aquasecurity/trivy/pkg/commands.NewAWSCommand.func2(0xc002f0cc80, {0xc0031c02d0?, 0x3?, 0x3?})
	/Users/saso/.ghq/github.com/aquasecurity/trivy/pkg/commands/app.go:844 +0x1bb
github.com/spf13/cobra.(*Command).execute(0xc002f0cc80, {0xc0031c02a0, 0x3, 0x3})
	/Users/saso/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc002669400)
	/Users/saso/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/saso/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.run()
	/Users/saso/.ghq/github.com/aquasecurity/trivy/cmd/trivy/main.go:37 +0x17c
main.main()
	/Users/saso/.ghq/github.com/aquasecurity/trivy/cmd/trivy/main.go:19 +0x19
```

> There were 1 errors during adaption process: failed to run adapter for api-gateway: operation error ApiGatewayV2: GetDomainNames, https response error StatusCode: 200, RequestID: 8d77a773-1d30-46e9-8bcd-811d4db6efa0, deserialization failed, failed to decode response body with invalid JSON, expected __timestampIso8601 to be of type string, got json.Number instead
panic: runtime error: invalid memory address or nil pointer dereference

## Output of `trivy -v`:

```
> ./trivy -v
Version: 0.30.0-73-gd343d13a
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-08-16 00:12:45.333566227 +0000 UTC
  NextUpdate: 2022-08-16 06:12:45.333565727 +0000 UTC
  DownloadedAt: 2022-08-16 00:41:09.323096 +0000 UTC
```

## Additional details (base image name, container registry info...):

--- a/go.mod
+++ b/go.mod
@@ -162,7 +162,7 @@ require (
 	github.com/alicebob/gopher-json v0.0.0-20200520072559-a9ecdc9d1d3a // indirect
 	github.com/apparentlymart/go-cidr v1.1.0 // indirect
 	github.com/apparentlymart/go-textseg/v13 v13.0.0 // indirect
-	github.com/aquasecurity/defsec v0.71.7
+	github.com/aquasecurity/defsec v0.71.8
 	github.com/asaskevich/govalidator v0.0.0-20200428143746-21a406dcc535 // indirect
 	github.com/aws/aws-sdk-go v1.44.77
 	github.com/beorn7/perks v1.0.1 // indirect

--- a/go.mod
+++ b/go.mod
@@ -162,7 +162,7 @@ require (
 	github.com/alicebob/gopher-json v0.0.0-20200520072559-a9ecdc9d1d3a // indirect
 	github.com/apparentlymart/go-cidr v1.1.0 // indirect
 	github.com/apparentlymart/go-textseg/v13 v13.0.0 // indirect
-	github.com/aquasecurity/defsec v0.71.8
+	github.com/aquasecurity/defsec v0.71.9
 	github.com/asaskevich/govalidator v0.0.0-20200428143746-21a406dcc535 // indirect
 	github.com/aws/aws-sdk-go v1.44.77
 	github.com/beorn7/perks v1.0.1 // indirect

--- a/pkg/cloud/aws/commands/run.go
+++ b/pkg/cloud/aws/commands/run.go
@@ -150,8 +150,9 @@ func Run(ctx context.Context, opt flag.Options) error {
 				for _, e := range aerr.Errors() {
 					log.Logger.Warnf("Adapter error: %s", e)
 				}
+			} else {
+				return fmt.Errorf("aws scan error: %w", err)
 			}
-			return fmt.Errorf("aws scan error: %w", err)
 		}
 		r = report.New(cloud.ProviderAWS, opt.Account, opt.Region, results.GetFailed(), opt.Services)
 	} else {

